pilot/pkg/xds/eds.go

```diff
@@ -418,7 +418,7 @@ func (eds *EdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w
 				Resource: util.MessageToAny(l),
 			}
 			resources = append(resources, resource)
-			eds.Server.Cache.Add(builder, token, resource)
+			eds.Server.Cache.Add(builder, model.CacheToken(req.Start.Nanosecond()), resource)
```

Comment:
nit: maybe Add should take a PushRequest object so each user doesn't need to know it must use req.Start.Nanosecond?

Reply:
Also we will need to do some refactoring in CDS to pass push request through for https://github.com/istio/istio/pull/33338, and in the future lds/rds. Shouldn't be a problem, just a note

Reply:
OK, in case others miss the intention here.